packages/jest-config/src/normalize.ts

```diff
@@ -46,19 +46,29 @@ type AllOptions = Config.ProjectConfig & Config.GlobalConfig;
 const createConfigError = (message: string) =>
   new ValidationError(ERROR, message, DOCUMENTATION_NOTE);
 
-const mergeOptionWithPreset = (
+// TS 3.5 forces us to split these into 2
```

Comment:
@jeysal is this something you know how to fix? 😀

Reply:
worked without split on 3.4 (I forgot to run `yarn`).

This _wonderful_ error on 3.5
![image](https://user-images.githubusercontent.com/1404810/60170407-ab250080-9808-11e9-8f52-4adf21d1c7a7.png)

Essentially, since the index signature does not match, it gets super confused even though we'll only be merging 1 at a time